packages/scaffold-config/src/frameworks.ts

```diff
@@ -14,10 +14,42 @@ export type WizardBundler = typeof dependencies.WIZARD_BUNDLERS[number]
 
 export type CodeGenFramework = Cypress.ResolvedComponentFrameworkDefinition['codeGenFramework']
 
+const yarnPnpRegistrationPath = new Map<string, { usesYarnPnP: boolean }>()
+
+async function readPackageJson (packageFilePath: string, projectPath: string): Promise<PkgJson> {
+  if (yarnPnpRegistrationPath.get(projectPath)?.usesYarnPnP) {
+    // using Yarn PnP. You cannot `fs.readJson`, since the module is zipped.
+    // use require.resolve - The PnP runtime (.pnp.cjs) automatically patches Node's
+    // fs module to add support for accessing files inside Zip archives.
+    // @see https://yarnpkg.com/features/pnp#packages-are-stored-inside-zip-archives-how-can-i-access-their-files
+    return require(require.resolve(packageFilePath))
+  }
+
+  return await fs.readJson(packageFilePath)
+}
+
 export async function isDependencyInstalled (dependency: Cypress.CypressComponentDependency, projectPath: string): Promise<Cypress.DependencyToInstall> {
   try {
     debug('detecting %s in %s', dependency.package, projectPath)
 
+    // we only need to register this once, when the project check dependencies for the first time.
+    if (!yarnPnpRegistrationPath.has(projectPath)) {
+      try {
+        const pnpapi = require(path.join(projectPath, '.pnp.cjs'))
```

Comment:
What about a scenario where a Cypress project is nested into a subdirectory of a yarn project? Do we need to traverse upwards to find the first `.pnp.cjs` we encounter?

Theres a method on the [`module` module](https://yarnpkg.com/advanced/pnpapi#requiremodule) that seems to do this for us in a PnP context (find `.pnp.cjs`, load, and return `pnpapi` (which we can ignore))

Reply:
Ah, forgot the nested edge case, I will verify what happens and address using this if required. Yarn has a ton of APIs for this, as you pointed out, we can probably use one.

Reply:
Painful, but for this API to exist you need to be in a PnP context... which you get by requiring the file in the first place 🤦 

I think we need to manually traverse up, lucky we've already got some code that does that, `isRepositoryRoot`.